### PR TITLE
Added building afl++ to docker container.

### PR DESCRIPTION
**Pull request by kobrineli**

### CODE REVIEW

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile [hunk 1]
```diff
@@ -33,4 +33,24 @@ RUN git clone https://github.com/llvm/llvm-project.git && cd llvm-project && \
 RUN echo /usr/lib/x86_64-unknown-linux-gnu > /etc/ld.so.conf.d/libc++.conf && \
     ldconfig
 
+RUN apt-get -y install lcov cargo pkg-config libglib2.0-dev \
```
Comment (SweetVishnya) [inline docker/ubuntu20.04-sydr-fuzz/Dockerfile:36]: Better install everything in upper apt-get install command

Comment (SweetVishnya) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile:36]: You may install rust (cargo) the following way:

```
# Install Rust.
RUN curl https://sh.rustup.rs -o rustup.sh && chmod +x rustup.sh && \
    ./rustup.sh -y && rm rustup.sh

ENV PATH="/root/.cargo/bin:${PATH}"
```

Comment (SweetVishnya) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile:36]: Then remove these lines from image-rs Dockerfile and verify that it works.

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile [hunk 2]
```diff
@@ -33,4 +33,24 @@ RUN git clone https://github.com/llvm/llvm-project.git && cd llvm-project && \
...
+    gcc-10 g++-10 gcc-10-plugin-dev gcc-10-multilib
+
+RUN cd / && git clone https://github.com/AFLplusplus/AFLplusplus.git && cd AFLplusplus && \
```
Comment (SweetVishnya) [inline docker/ubuntu20.04-sydr-fuzz/Dockerfile:39]: You are already in /

Comment (SweetVishnya) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile:39]: You may clone afl++ in the same RUN command that builds it. Do not forget to remove afl++ sources directory later.

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile [hunk 3]
```diff
@@ -33,4 +33,24 @@ RUN git clone https://github.com/llvm/llvm-project.git && cd llvm-project && \
...
+    git checkout a790bf6cc2bb63f0659d8c96b46f5f4db2f9d4bc
+
+RUN git clone --depth=1 https://github.com/vanhauser-thc/afl-cov /afl-cov && cd /afl-cov && \
+    git checkout bb51de02c06dca2bfa2dfb494fa3b095950ec879 && \
+    make install && cd ..
```
Comment (SweetVishnya) [inline docker/ubuntu20.04-sydr-fuzz/Dockerfile:44]: Enable `-j 200` here and for afl++.

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile [hunk 4]
```diff
@@ -33,4 +33,24 @@ RUN git clone https://github.com/llvm/llvm-project.git && cd llvm-project && \
...
+ENV LLVM_CONFIG=llvm-config
+ENV AFL_SKIP_CPUFREQ=1
+
+WORKDIR /AFLplusplus
+
+RUN export CC=gcc-10 && export CXX=g++-10 && make clean && \
+    make distrib && make install && make clean
+
 ENV PATH=$PATH:/fuzz/sydr
+
```
Comment (SweetVishnya) [inline docker/ubuntu20.04-sydr-fuzz/Dockerfile:55]:
```
ENV IS_DOCKER="1"
```

This may be useful

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile (hunk 4 above)
Comment (kobrineli) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile:58]: Done

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile (hunk 3 above)
Comment (kobrineli) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile]: Done

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile (hunk 2 above)
Comment (kobrineli) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile]: Done

File: docker/ubuntu20.04-sydr-fuzz/Dockerfile (hunk 1 above)
Comment (kobrineli) [reply docker/ubuntu20.04-sydr-fuzz/Dockerfile]: Done

**Pull request merged**